src/gui/SubWindow.cpp

```diff
@@ -301,25 +366,43 @@ void SubWindow::attach()
 
 	if (!isDetached()) { return; }
 
-	auto frame = widget()->windowHandle()->frameGeometry();
+	const bool shown = widget()->isVisible();
+
+	auto frame = widget()->windowHandle()->geometry();
+	frame.moveTo(mdiArea()->mapFromGlobal(frame.topLeft()));
+	frame += decorationMargins();
+
+	// arbitrary values, require window to touch `mdiArea - margin`
+	// min(max(a, b), c) ensures that a <= b <= c
+	// TODO make this live configurble maybe?
+	const auto margin = QMargins(40, 40, 40, 40);
+	frame.moveTo(std::min(std::max(margin.left() - frame.width(),
```

Comment:
Use `std::clamp` instead? And does `QMargins` bring some value or does it just make the code longer?